addons/mail/static/src/discuss/core/common/discuss_notification_settings.js

```diff
@@ -23,6 +23,10 @@ export class DiscussNotificationSettings extends Component {
         }
     }
 
+    onChangeMessageSound() {
+        this.store.settings.messageSound = !this.store.settings.messageSound;
```

Comment:
Anything "toggle" is not a good design, if t-model it not possible it should at least read the current value of the view rather than make assumption based on previous value.